packages/entity-server/src/routes/hierarchy/EntityDescendantsRoute.ts

```diff
@@ -0,0 +1,25 @@
+/**
+ * Tupaia
+ * Copyright (c) 2017 - 2021 Beyond Essential Systems Pty Ltd
+ */
+
+import { Route } from '../Route';
+import {
+  HierarchyRequest,
+  HierarchyResponse,
+  HierarchyRequestParams,
+  EntityResponseObject,
+} from './types';
+
+export class EntityDescendantsRoute extends Route<
+  HierarchyRequest<HierarchyRequestParams, EntityResponseObject[]>,
+  HierarchyResponse<EntityResponseObject[]>
+> {
+  async buildResponse() {
+    return this.res.context.formatEntitiesForResponse(
+      [this.req.context.entity].concat(
+        await this.req.context.entity.getDescendants(this.req.context.hierarchyId),
+      ),
```

Comment:
Attaching the requested entity to it's descendant tree here... is this expected? It's inline with what we currently do in `web-config-server` and it does have the upside of saving the client from doing multiple requests to get a root entity and it's descendants (which is quite common) but I can see how it may be unanticipated...

Reply:
I like this suggestion :)